numpy/core/arrayprint.py

```diff
@@ -613,6 +601,8 @@ def __init__(self, data, precision, suppress_small, sign=False):
         self.exp_format = False
         self.large_exponent = False
         self.max_str_len = 0
+        self.nan_str = _options['nanstr']
+        self.inf_str = _options['infstr']
```

Comment:
This is a behaviour change - it means that after the formatter is created,  changing the nanstr has no effect

Reply:
Brilliant. And thanks. I never thought about that.

Reply:
In case it's not clear, what I'm saying here is that you need to undo these lines, because a maintenance PR shouldn't be changing behaviour.

Reply:
i got it. just that i didn't get the time to redo it.